### ISSUE TITLE
[CLI]: ERROR Internal wandb error: file data was not synced When using sync_tensorboard with Jupyter notebook

### ISSUE THREAD

**Issue by BastiaanBergman** [wandb/wandb#7792]
### Describe the bug

Running WandB from a Jupyter notebook often works but not in combination with the sync_tensorboard option. sync_tensorboard does work from a commandline python script. Just the combination sync_tensorboard/Jupyter is giving me errors.

```python
from torch.utils.tensorboard import SummaryWriter
import wandb

wandb.login(key='xxx')
run = wandb.init(project="x", sync_tensorboard=True)
tboard = SummaryWriter()
tboard.add_scalar('Loss/train', 12, 13)
```

```shell
Thread HandlerThread:
Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.11/site-packages/traitlets/traitlets.py", line 653, in get
    value = obj._trait_values[self.name]
            ~~~~~~~~~~~~~~~~~^^^^^^^^^^^
KeyError: 'layout'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/internal/internal_util.py", line 48, in run
    self._run()
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/internal/internal_util.py", line 99, in _run
    self._process(record)
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/internal/internal.py", line 278, in _process
    self._hm.handle(record)
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/internal/handler.py", line 150, in handle
    handler(record)
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/internal/handler.py", line 786, in handle_tbrecord
    self._tb_watcher.add(tbrecord.log_dir, tbrecord.save, tbrecord.root_dir)
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/internal/tb_watcher.py", line 166, in add
    self._consumer = TBEventConsumer(
                     ^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/internal/tb_watcher.py", line 373, in __init__
    self._internal_run = internal_run.InternalRun(run_proto, settings, datatypes_cb)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/internal/run.py", line 15, in __init__
    super().__init__(settings=settings)
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/wandb_run.py", line 594, in __init__
    self._init(
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/wandb_run.py", line 630, in _init
    self._printer = get_printer(self._settings._jupyter)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/lib/printer.py", line 312, in get_printer
    return PrinterJupyter()
           ^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/lib/printer.py", line 231, in __init__
    self._progress = ipython.jupyter_progress_bar()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/lib/ipython.py", line 144, in jupyter_progress_bar
    return ProgressWidget(widgets, min=min, max=max)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/wandb/sdk/lib/ipython.py", line 103, in __init__
    self._progress = widgets.FloatProgress(min=min, max=max)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/ipywidgets/widgets/widget_float.py", line 26, in __init__
    super(_Float, self).__init__(**kwargs)
  File "/opt/anaconda3/lib/python3.11/site-packages/ipywidgets/widgets/widget.py", line 480, in __init__
    self.open()
  File "/opt/anaconda3/lib/python3.11/site-packages/ipywidgets/widgets/widget.py", line 493, in open
    state, buffer_paths, buffers = _remove_buffers(self.get_state())
                                                   ^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/ipywidgets/widgets/widget.py", line 591, in get_state
    value = to_json(getattr(self, k), self)
                    ^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/traitlets/traitlets.py", line 700, in __get__
    return self.get(obj, cls)
           ^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/traitlets/traitlets.py", line 656, in get
    default = obj.trait_defaults(self.name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/traitlets/traitlets.py", line 1868, in trait_defaults
    return self._get_trait_default_generator(names[0])(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/traitlets/traitlets.py", line 624, in default
    return self.make_dynamic_default()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/ipywidgets/widgets/trait_types.py", line 168, in make_dynamic_default
    return self.klass(*(self.default_args or ()),
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/ipywidgets/widgets/widget.py", line 480, in __init__
    self.open()
  File "/opt/anaconda3/lib/python3.11/site-packages/ipywidgets/widgets/widget.py", line 511, in open
    self.comm = create_comm(**args)
                ^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/site-packages/comm/__init__.py", line 27, in _create_comm
    raise NotImplementedError("Cannot ")
NotImplementedError: Cannot 
wandb: ERROR Internal wandb error: file data was not synced
```

### Environment

WandB version: wandb-0.17.1

OS: Mac Sonoma 14.3

Python version: 3.11.7 (main, Dec 15 2023, 12:09:56) [Clang 14.0.6 ]

Versions of relevant libraries: pytorch: 2.3.0, traitlets-5.14.3, Jupyter notebook Version: 7.0.8

**Comment by paulosabile-wb** [comment]
Hi @BastiaanBergman Good day and thank you for reporting this issue to us. Happy to help here.

Could you please provide us the `debug-internal.log` and `debug.log` for the affected run. These files are under your local folder `wandb/run-_-/logs` in the same directory where you’re running your script. These files will help us with more details about this error and we can further investigate.

**Comment by BastiaanBergman** [comment]
Yes, please see attached:

**Comment by paulosabile-wb** [comment]
Hi @BastiaanBergman Good day! I am not seeing any files attached on this thread. Is it possible if you can attached the request files on this zendesk link? 
https://weightsandbiases.zendesk.com/agent/tickets/69113

**Comment by BastiaanBergman** [comment]
logs are attached here now, does that work?
[debug-internal.log](https://github.com/user-attachments/files/15897270/debug-internal.log)
[debug.log](https://github.com/user-attachments/files/15897271/debug.log)

**Comment by paulosabile-wb** [comment]
Hi @BastiaanBergman 

Good day and thank you for patiently waiting for my update. We reviewed this and we'd like to ask you to try to terminate the run with wandb.finish(). This will finish the wandb run and upload the tensorboard logs to W&B so they can be visualized. More information [here](https://docs.wandb.ai/guides/integrations/tensorboard#google-colab-jupyter-and-tensorboard).

Let me know if this solves the issue.


Thanks,
Paulo

**Comment by BastiaanBergman** [comment]
Paulo, Thank you for your help with this. I don't understand the proposed solution here, `wandb.finish()` will be called only at the end, correct? The error happens before we reach the end, and it terminates further execution.

**Comment by paulosabile-wb** [comment]
Hi @BastiaanBergman Good day! You are correct, wandb.finish() should be called at the end of your training/code. Could you provide us a code snippet or a sample toy code that would reproduce this error that you are getting in your Jupyter Notebook so we can review? We appreciate your efforts working with us for this issue. I am looking forward for your next update!

Best Regards,
Paulo

**Comment by BastiaanBergman** [comment]
Paulo appears to be a bot.

**Comment by paulosabile-wb** [comment]
Hi @BastiaanBergman Apologies for this. Please disregard my last update, I missed to check that the code was provided on the first part of the thread. I will further check this with the team and I will get back to you for an update.

**Comment by paulosabile-wb** [comment]
Hi @BastiaanBergman Could you please try to update your Jupyter notebook version to the latest version and test?

Also, can you confirm if you see this issue only when running their code in a Jupyter env, or if the same happens for you in other environment like Colab, or when executing the script in your terminal?

**Comment by BastiaanBergman** [comment]
No, sorry, I cannot do that. Best of luck.

**Comment by sephmard** [comment]
Hey @BastiaanBergman, PM for W&B here. We recently released a new Go-based client for W&B that includes some stability improvements, which might address the TB sync issue that you are encountering. You can try it out now by adding wandb.require("core") to your scripts. The updated client is available here: https://wandb.me/wandb-core. This will be released as the default SDK backend with v0.18.0. 

Please give it a try now and let us know if it does not resolve this issue.
